solver/cvrp.py

```diff
@@ -437,31 +406,33 @@ def check_feasibility(solution):
 
 
         num_states = model.states.size()
-        solutions = []
+        solution = []
         for i in range(num_states):
-            # extract the solution from the route NL variables
-            solution = [[int(v) + 1 for v in route.state(i)] for route in routes]
+            # extract the solution
+            decision = next(model.iter_decisions())
+            solution_candidate = [[int(v) + 1 for v in route.state(i)] for route in list(decision.iter_successors())]
 
             solver_objective = model.objective.state(i)
-            assert abs(solver_objective - recompute_objective(solution)) < tolerance
+            assert abs(solver_objective - recompute_objective(solution_candidate)) < tolerance
```

Comment:
Good catch, I've added a check for this and put everything in a function because I think it makes more sense than breaking. Let me know if this makes sense: https://github.com/dwave-examples/mvrp/pull/12/commits/2e06b8278c86fc5c4f3df9368cb4f3f0318df5d7